### ISSUE TITLE
[REVIEW]: CoPro: a data-driven model for conflict risk projections

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2855]
**Submitting author:** @JannisHoch (<a href="http://orcid.org/0000-0003-3570-6436">Jannis Hoch</a>)
**Repository:** <a href="https://github.com/JannisHoch/copro" target ="_blank">https://github.com/JannisHoch/copro</a>
**Version:** v0.0.5
**Editor:** @sbenthall
**Reviewer:** @soodoku, @sbenthall
**Archive:** <a href="https://doi.org/10.5281/zenodo.4548904" target="_blank">10.5281/zenodo.4548904</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/1f03334e56413ff71f65092ecc609aa4/status.svg)](https://joss.theoj.org/papers/1f03334e56413ff71f65092ecc609aa4)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/1f03334e56413ff71f65092ecc609aa4"><img src="https://joss.theoj.org/papers/1f03334e56413ff71f65092ecc609aa4/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/1f03334e56413ff71f65092ecc609aa4/status.svg)](https://joss.theoj.org/papers/1f03334e56413ff71f65092ecc609aa4)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/JannisHoch/copro" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@soodoku & @sbenthall, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @sbenthall know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JannisHoch/copro">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@JannisHoch) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JannisHoch/copro">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@JannisHoch) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @soodoku, @sbenthall it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by sbenthall** [comment]
@soodoku Sorry for the delay on starting the review. Now the review has started, and you can see the checklist on this issue.

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1177/0022343317691330 is OK
- 10.1177/0022343316682065 is OK
- 10.1177/0022343316682064 is OK

MISSING DOIs

- 10.1126/science.aal4483 may be a valid DOI for title: Predicting armed conflict: Time to adjust our expectations?
- 10.1088/1748-9326/11/5/054002 may be a valid DOI for title: Forecasting civil conflict along the shared socioeconomic pathways

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02855/joss.02855/10.21105.joss.02855.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02855/joss.02855/10.21105.joss.02855.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
:wave: @soodoku, please update us on how your review is going.

**Comment by whedon** [comment]
:wave: @sbenthall, please update us on how your review is going.

**Comment by JannisHoch** [comment]
HI @soodoku and @sbenthall, in the meantime, the model evolved to version 0.0.6 (https://github.com/JannisHoch/copro/releases/tag/v0.0.6). All changes made in the process of this review will be committed to the branch 'joss-review' (https://github.com/JannisHoch/copro/tree/joss_review).

**Comment by soodoku** [comment]
i am done w/ my review.

**Comment by sbenthall** [comment]
I apologize for the delay in my review.
There have been some personal issues on my end that have needed extra attention.
I'm proceeding with my review now.

**Comment by sbenthall** [comment]
I think this is terribly exciting and I am eager to see it published in the best form possible. To that end:

* Please document the requirement that Conda needs to be installed to run the installation instructions.
* Is there any other comparable software that should be considered as the 'state of the field' in this line of research?
* I understand that the library has been designed for flexibility. However, I am having trouble determining from the paper and indeed from the documentation what the core functionality of the library is. I wonder if it would be possible to foreground a bit more technical detail about what scaffolding the library is doing for modeling work, and what kinds of models are included out-of-the-box.

I have not yet completed reviewing the code--these are first impressions of the document and paper only. But I think these improvements to the writeup and documentation are important for the publication.

**Comment by sbenthall** [comment]
When I get to the `sh run.sh` part of the instructions in the README, I get the following error:

https://gist.github.com/sbenthall/83de0ce8675283c01d3f3fa973dc6f38

**Comment by JannisHoch** [comment]
Hi @sbenthall,
Absolutely no problem!

From your comments I assume you were not looking at the most recent version of the model.
For instance, in the joss-review branch (https://github.com/JannisHoch/copro/tree/joss_review), there is no file run.sh anymore. It is renamed to 'run_notebooks.sh' for running the jupyter notebooks, and 'run_script_*.sh' for executing the command line scripts.

If your problems still occur after installing the latest copro version, please let me know. thanks :)

**Comment by JannisHoch** [comment]
in f9326d36bcbf20e93a54e6ea5733f1ae515f4477 I have added notes that the model was developed and tested with anaconda/miniconda.

**Comment by JannisHoch** [comment]
regarding your question of other state-of-the-art software: to my knowledge, there is no freely available software model with such an extensive documentation out there. of course there is model code as supplement to a publication, and publications covering similar approaches (see the references in the joss manuscript), but not in a 'structured' way like done for CoPro.

**Comment by JannisHoch** [comment]
I also have added a bit more technical detail to the JOSS manuscript. based on the review of @soodoku, the model goal and purpose were made clearer in the README and in the documentation.

**Comment by sbenthall** [comment]
Hello. Apologies for the further delay. I've had to take some family leave but am now back to work. I'll look at the branch you've referenced; sorry to miss it the first time.

**Comment by sbenthall** [comment]
I am being held up by some local Python environment issues. (The `rasterio` dependency is not installing properly.) I'm pretty sure this is just a problem on my end! I'll try to repair my environment setup and get back to you.

**Comment by JannisHoch** [comment]
@sbenthall, first of all, good to hear you're back!

Could you maybe attach a screenshot or so for me to look into it?

And I assume you were using the yml file to create the environment?

**Comment by sbenthall** [comment]
This is the stack trace. https://gist.github.com/sbenthall/5edad4e128fea79a99c2486ab6333bfe
Googling for it makes it look like a local python issue. I'll keep kicking at it.

**Comment by sbenthall** [comment]
Hmm.
So what's strange here is that `conda install rasterio==1.1.0` works fine, but then I get an error when it installs again via `pip install -e .`.

I wonder if the conda version is conflicting with the pip version somehow (the pip version is invoked in the `setup.py` file.)

**Comment by sbenthall** [comment]
Commenting out the `rasterio` dependency in `setup.py` and rerunning the `pip` installation gets me this error:

https://gist.github.com/sbenthall/6a7f135970232dad08eb65ecebfd97cf

**Comment by sbenthall** [comment]
I do notice that `rasterio` installation instructions look like they require a number of other GDAL related packages, which in the `rasterio` installation instructions are listed as separate steps.

https://rasterio.readthedocs.io/en/latest/

I wonder if you could confirm that your instructions for CoPro, as written, work on a fresh system that does not have these GDAL packages installed. It is quite likely that these are being handled correctly by Anaconda and there's some issue going on locally, but if you could test on another machine that would be additional confirmation that it's a local issue and not a problem with the installation instructions.

**Comment by JannisHoch** [comment]
Hmmm, this all sounds pretty odd to me.

I have never seen any of the error messages before when installing copro - neither the one when running `pip install -e .` (as it seems that rasterio is correctly installed from cache and the error occurs bit later somehow), nor the one with altered `setup.py` file (which is a permission error after all if I get it right?).

Also, I was able to install copro (branch joss_review) on a pretty 'fresh' Linux environment.

So yes, it may be that it's a local issue. I have two ideas:

- use a different linux/windows enviroment?
- in the `environment.yml` file, put the rasterio dependency in the pip section instead the conda-forge section and see if this solves the suspected conflict between conda/pip?

**Comment by JannisHoch** [comment]
Or, as another possible fix, specify a more recent rasterio version? current one is 1.1.8. maybe there are some flaws in version 1.1.0 for some specific circumstances.

**Comment by sbenthall** [comment]
Good news. I've got the install working. Sorry about that.

Now I'm trying to test the library.
Tiny snag: downloading the demo data and putting it in the right place so that the .cfg files can find it requires a little trial and error for me at the moment.

Certainly not a dealbreaker but some scripting around that would be nice.

I'll continue my review over the weekend -- it should be done by Monday. Thanks for your patience on this--I know it's been a long wait.

**Comment by JannisHoch** [comment]
great to hear - i am curious, what exactly was the problem?

so the 'download_example_data.sh' does not what it's supposed to do? guess i'll here that from you soon.
but to not slow down the review process, i could also send you the example data via mail?

looking forward to the remaining review!

**Comment by sbenthall** [comment]
> so the 'download_example_data.sh' does not what it's supposed to do?

oh! I didn't see this. It is not mentioned in the README --- it just says the data is on Zenodo. So I fetched it by hand...

**Comment by sbenthall** [comment]
https://copro.readthedocs.io/en/latest/examples/nb02_XY_data.ipynb.html 
4.2.2.2. "Assing" -> "Assign"

**Comment by sbenthall** [comment]
I see your example notebooks use `%matplotlib inline`. I believe more recent versions of Jupyter have made this unnecessary.

**Comment by sbenthall** [comment]
As you direct users to use your Jupyter notebook examples, it would be helpful if you referred them to instructions for setting up their Jupyter notebooks to be able to use the conda environment as a kernel. e.g.

https://medium.com/@nrk25693/how-to-add-your-conda-environment-to-your-jupyter-notebook-in-just-4-steps-abeab8b8d084

**Comment by sbenthall** [comment]
nb01 cell [5] I'm getting the following error:

`conflict_gdf, extent_gdf, selected_polygons_gdf, global_df = selection.select(config, out_dir, root_dir)`
https://gist.github.com/sbenthall/24da689b34fef8fc2ef9e16b99027215

nb02 cell 3 I'm getting the following error:
`conflict_gdf = gpd.read_file('conflicts.shp')
selected_polygons_gdf = gpd.read_file('polygons.shp')`
https://gist.github.com/sbenthall/d8a722add82fa013d3779e39953773fe

....

I see now that these shapefiles are created by n01 in normal operation, and that the `run_notebooks.sh` script cleans these out. But I'm not deep enough in to know why nb01 is failing when ran manually.

....

I know it's hard to get the tooling around this perfectly but I think with a little more documentation the initial user experience would be more idiot-proof.

**Comment by JannisHoch** [comment]
hi @sbenthall ,

thanks for your comments. Regarding the first three, I will change them asap.

Regarding the last, few commetns from my side.

-  seems like you don't have the package `rtree` installed. I checked the `environment.yml` file and it is indeed missing there. but thus far i was under the impression that this was installed along `rasterio` - maybe your initial installation issues can explain this error. either way, i will add `rtree` to the yml-file.
- indeed, some files are written to file in nb01 and then loaded in the other notebooks. this is because when initializing the model, all output files are removed if the output folder already exists. so that is a way to circumvene this 'issue' when using notebooks. note, when using the command line script, this problem of course does not persist!
- the error you get in nb02 is thus related to the one in nb01. if you cannot run nb01 succesfully, the remaining notebooks won't work either since some files are missing!
- i'll work on the documentation in the notebooks asap.

**Comment by JannisHoch** [comment]
I have added `rtree` and `ipykernel` to the `environment.yml` file, so hope that a) writing the files in nb01 works now, and b) adding the copro-environmen to your jupyter kernel is possible.

Also, I have fixed the typo, added info about the `download_example_data.sh` file in `README.md`, and placed contextual info on the execution of the notebooks there as well.

**Comment by sbenthall** [comment]
Ok, just tried the `.sh` scripts version of working README.

I ran `sh run_script_reference.sh` no problem.

Then with `sh run_script_projections.sh` I got this error:

https://gist.github.com/sbenthall/11fdd17af6f690c13933383e504eac50

**Comment by JannisHoch** [comment]
hi @sbenthall, 
I am again stligthly puzzled about the errror you get.
If I run the scripts in the 'joss_review' branch (which is the one under review I guess?), I don't get any issues at all.
Also, I get different console output than you.

sure you use the right branch?

**Comment by JannisHoch** [comment]
dear @sbenthall ,
hope you are doing well.
I am just curious how the review process is going on your side.
Not wanna be pushy as I know that this work is voluntary and Covid-19 is not doing any good either...
Since I want to use the code in a manuscript, I'd very much appreciate it if there could be a citable version of the code until then.
Thanks!
Looking forward to any new bugs you find.
Jannis

**Comment by sbenthall** [comment]
I apologize again for the delay. We had a baby in December.
I'm double checking this today

**Comment by sbenthall** [comment]
I'm using the `joss_review` branch and am getting the same error.

What is your output for that script?

**Comment by JannisHoch** [comment]
that's great news, congrats!

i am bit puzzled by your error message, or more precisely by the console output you get.
it does not look like anything i get when using the `joss_review` branch with the cfg-files as is plus the downloaded example data.
it seems like you getting debug output, but not the one i get using the `joss_review` branch.
that is what makes me assume that there is something odd about your model exeuction, but cannot pinpoint what it is.

did you alter any of the files by any chance?
maybe good if you could send me the cfg-files you use if the error persists?

**Comment by sbenthall** [comment]
I'll try a fresh install

**Comment by sbenthall** [comment]
It works!
Thanks for your patience.
This passes review and we'll proceed to the next stage.

**Comment by sbenthall** [comment]
@whedon check references

**Comment by sbenthall** [comment]
@whedon generate pdf

**Comment by JannisHoch** [comment]
@sbenthall so what was the magic trick? anything i can add to the package to make the installation bullet-proof?

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02855/joss.02855/10.21105.joss.02855.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02855/joss.02855/10.21105.joss.02855.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1177/0022343317691330 is OK
- 10.1177/0022343316682065 is OK
- 10.1177/0022343316682064 is OK

MISSING DOIs

- 10.1126/science.aal4483 may be a valid DOI for title: Predicting armed conflict: Time to adjust our expectations?
- 10.1088/1748-9326/11/5/054002 may be a valid DOI for title: Forecasting civil conflict along the shared socioeconomic pathways

INVALID DOIs

- None
```

**Comment by sbenthall** [comment]
It was likely just human error initially. Don't worry about it.

Next steps:
- proofread the paper and make sure there are no typos.
- Double check the MISSING DOIs that @whedon brought up
 
This is the final editing stage.

**Comment by JannisHoch** [comment]
am i supposed to push any changes to the `joss_review` branch or how does this work?

**Comment by sbenthall** [comment]
If you have edits to the paper, then yes, that's right.

When it's absolutely finalized, you tag it.

**Comment by sbenthall** [comment]
Yeah, those DOI's Whedon pointed out should be added to the joss .bib file

**Comment by sbenthall** [comment]
"a key concept in mind when establishing CoPro was to be able to produce reproducible output using transparent models"

grammar issue ... "a concept" "was" "to be able to"

suggested revision: "CoPro produces reproducible output using transparent models."

**Comment by sbenthall** [comment]
I'm wondering about the title "a data-driven model for conflict risk projections" .
From the article text, it sounds like CoPro is a toolkit allowing for many different models?

**Comment by JannisHoch** [comment]
this depends how one defines model, a maybe bit philosophical question.
so i'd say that CoPro is a model in itself, but makes use of other models again... so maybe call it a modelling framework?

**Comment by JannisHoch** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by JannisHoch** [comment]
the version number is 0.0.7 and the zenodo DOI is https://doi.org/10.5281/zenodo.4548904.

**Comment by JannisHoch** [comment]
that's how it's supposed to work? or do I need to let wheedon update version number and DOI? not fully clear to me from the instructions i must admit.

**Comment by JannisHoch** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1177/0022343317691330 is OK
- 10.1177/0022343316682065 is OK
- 10.1177/0022343316682064 is OK

MISSING DOIs

- 10.1126/science.aal4483 may be a valid DOI for title: Predicting armed conflict: Time to adjust our expectations?
- 10.1088/1748-9326/11/5/054002 may be a valid DOI for title: Forecasting civil conflict along the shared socioeconomic pathways

INVALID DOIs

- None
```

**Comment by JannisHoch** [comment]
hmmm whedon seems to refer to an old version...

**Comment by sbenthall** [comment]
That's weird. Maybe whedon needs to regenerate the PDF before checking references again

**Comment by sbenthall** [comment]
@whedon generate pdf

**Comment by sbenthall** [comment]
@whedon check references

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02855/joss.02855/10.21105.joss.02855.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02855/joss.02855/10.21105.joss.02855.pdf) :page_facing_up: :point_left:

**Comment by arfon** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1177/0022343317691330 is OK
- 10.1177/0022343316682065 is OK
- 10.1177/0022343316682064 is OK

MISSING DOIs

- 10.1126/science.aal4483 may be a valid DOI for title: Predicting armed conflict: Time to adjust our expectations?
- 10.1088/1748-9326/11/5/054002 may be a valid DOI for title: Forecasting civil conflict along the shared socioeconomic pathways

INVALID DOIs

- None
```

**Comment by arfon** [comment]
@JannisHoch - are the latest changes on the `joss_review` branch?

**Comment by JannisHoch** [comment]
@arfon yes they are. And all put into a zenodo release already. But it's somehow not what wheedon is producing...

**Comment by arfon** [comment]
Right, so we need to be explicit with Whedon where to check (otherwise it just uses the default branch).

**Comment by arfon** [comment]
@whedon generate pdf from branch joss_review

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_review. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02855/joss.02855/10.21105.joss.02855.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02855/joss.02855/10.21105.joss.02855.pdf) :page_facing_up: :point_left:

**Comment by arfon** [comment]
@whedon check references from branch joss_review

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch joss_review
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1146/annurev-polisci-050317-070830 is OK
- 10.1038/s41586-019-1300-6 is OK
- 10.1126/science.aal4483 is OK
- 10.1088/1748-9326/11/5/054002 is OK
- 10.1177/0022343317691330 is OK
- 10.1177/0022343316682065 is OK
- 10.1177/0022343316682064 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.4548904 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4548904" target="_blank">10.5281/zenodo.4548904</a> is the archive.

**Comment by arfon** [comment]
@sbenthall @JannisHoch - I've updated the submission with the Zenodo archive. Next steps here are:

- Run `@whedon accept from branch joss_review `

Or, have @JannisHoch merge their changes into the default branch on their repository which means you then would need to run:

- `@whedon accept`

In both cases, the `@whedon accept` command will produce the final PDF proofs and notify the EiC on rotation that this submission is ready to accepted and published.

**Comment by JannisHoch** [comment]
@arfon 
many thanks for fixing this.

@arfon @sbenthall 
I have merged branch `joss_review` into default (which is `dev`, or does whedon choosing `master` as default by default?).

**Comment by arfon** [comment]
@JannisHoch - Whedon will use whatever the default is which sounds like it's `dev` so I think we should be good to go here.

**Comment by JannisHoch** [comment]
indeed, then we are good to go!

**Comment by sbenthall** [comment]
Thanks for your help @arfon I didn't know that about whedon and branches

**Comment by sbenthall** [comment]
@whedon accept from branch joss_review

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1146/annurev-polisci-050317-070830 is OK
- 10.1038/s41586-019-1300-6 is OK
- 10.1126/science.aal4483 is OK
- 10.1088/1748-9326/11/5/054002 is OK
- 10.1177/0022343317691330 is OK
- 10.1177/0022343316682065 is OK
- 10.1177/0022343316682064 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2091

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2091, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch joss_review 
```

**Comment by sbenthall** [comment]
@whedon accept deposit=true from branch joss_review

**Comment by whedon** [comment]
I'm sorry @sbenthall, I'm afraid I can't do that. That's something only editor-in-chiefs are allowed to do.

**Comment by sbenthall** [comment]
@JannisHoch Congratulations on your acceptance. Thanks for your patience with the (long!) reviewing process.

**Comment by sbenthall** [comment]
Oh, whoops. Overstepped my authority there! I guess we have to wait for an editor-in-chief...

**Comment by JannisHoch** [comment]
@sbenthall just to inform you, I have by now merged everything into dev and removed the joss_review branch no long3r exists... But if that's leading to no issues fine by me!

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1146/annurev-polisci-050317-070830 is OK
- 10.1038/s41586-019-1300-6 is OK
- 10.1126/science.aal4483 is OK
- 10.1088/1748-9326/11/5/054002 is OK
- 10.1177/0022343317691330 is OK
- 10.1177/0022343316682065 is OK
- 10.1177/0022343316682064 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2092

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2092, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1363453951430000640) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2099
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02855](https://doi.org/10.21105/joss.02855)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@soodoku, @sbenthall - many thanks for your reviews here and to @sbenthall for editing too! JOSS relies upon the volunteer efforts of people like you and we simply wouldn't be able to do this without you ✨ 

@JannisHoch - your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02855/status.svg)](https://doi.org/10.21105/joss.02855)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02855">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02855/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02855/status.svg
   :target: https://doi.org/10.21105/joss.02855
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02855/status.svg)](https://doi.org/10.21105/joss.02855)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
